packages/nuxt/src/core/nuxt.ts

```diff
@@ -308,6 +308,22 @@ async function initNuxt (nuxt: Nuxt) {
     addPlugin(resolve(nuxt.options.appDir, 'plugins/preload.server'))
   }
 
+  // Add type-checking
+  if (process.dev && nuxt.options.builder === '@nuxt/vite-builder' && nuxt.options.typescript.typeCheck === true) {
```

Comment:
I think this section should ideally be in the vite builder package but the plugin approach seems reasonable.

Reply:
Yeah, I tried to put it right after `addVitePlugin` part, but couldn't get it to work.
`addTemplate` does nothing at that point.

Reply:
Yes, that does make sense. 👍 I wonder if instead we could add the import to the entry in the vite client config?

Reply:
Updated the code. Hope I understood correctly.

Reply:
Thanks! Yes, perfect 👍